api/lib/subscription/slashing.js

```diff
@@ -38,20 +63,69 @@ class SlashingMonitor {
     )
 
     // requires some more logic to not spam the notifications if a validator is down for 1000 blocks
-    // this.client.subscribe(
-    //   { query: 'liveness.missed_blocks >= 1' },
-    //   (response) => {
-    //     const missedBlocks = response.events['liveness.address'].map(
-    //       (address, index) => ({
-    //         operatorAddress: address,
-    //         missedBlocks: response.events['liveness.missed_blocks'][index],
-    //         height: response.height
-    //       })
-    //     )
-    //     console.log('Missed block', missedBlocks)
-    //   }
-    // )
+    this.client.subscribe(
+      { query: `tm.event='NewBlockHeader' AND liveness.missed_blocks >= 1` },
+      async (response) => {
+        try {
+          const missedBlocks = response.events['liveness.address'].map(
+            (address, index) => ({
+              networkId: this.networkId,
+              operatorAddress: address,
+              blocks: response.events['liveness.missed_blocks'][index],
+              reason: 'missed_blocks',
+              height: response.data.value.header.height
+            })
+          )
+          const rows = await Promise.all(
+            missedBlocks.map(async (missedBlockEvent) => {
+              const lastMissedBlockEvent = await getLastMissedBlock(
+                this.networkId,
+                missedBlockEvent.operatorAddress
+              )
+              return aggregateMissedBlocks(
+                lastMissedBlockEvent,
+                missedBlockEvent
+              )
+            })
+          )
+          database(config)('').upsert('slashes', rows)
+          console.log('Wrote missed block events', rows.length)
+        } catch (error) {
+          console.error('Failed to write missed block events', error)
+          Sentry.captureException(error)
+        }
+      }
+    )
+  }
+}
+
+async function getLastMissedBlock(networkId, operatorAddress) {
+  // get last liveness issue
```

Comment:
```suggestion
async function getLastSlash(networkId, operatorAddress) {
  // get last slash
```

I don't know if I'm missing something here but this is what this query looks like is doing

Reply:
with this filter `reason:{_eq:"missed_blocks"}` I think the description is pretty decent